TypeScript.py

```diff
@@ -686,7 +687,11 @@ def on_close(self, view):
                # make sure we know the latest state of the file
                reloadBuffer(view, info.clientInfo)
                # notify the server that the file is closed
-               cli.service.close(view.file_name())
+               self.batch_close_queue.append(view.name())
+               if len(self.fileMap) <= 1 or len(self.batch_close_queue) = 5:
```

Comment:
Thanks for pointing out. And it turns out I need to batch the process in the server side also, so much more code changes needed to actually fix this. Will commit later.